### PR TITLE
Initial implemention of Student homepage

### LINKED ISSUE #102: Update Student homepage to show current Coop information

### PR DESCRIPTION
**Pull request by kraglalbert**
# Summary

This PR does an initial implementation of the Student's homepage. 

* Can view current Coop information
* Can view upcoming Coops and their status
* An alert shows if a Coop is approved but is missing CoopDetails

## Test Plan

Screen Shot 2020-03-29 at  Mar 29   6 10 52 PM
Screen Shot 2020-03-29 at  Mar 29   6 11 01 PM

## Related Issues

closes #102, begins #130

### CODE REVIEW

**Comment by emmaeagles** [comment]
LOL huge flex on the salary

File: Backend/src/main/java/ca/mcgill/cooperator/controller/StudentController.java [hunk 1]
```diff
@@ -204,6 +205,26 @@ public CoopDto getCurrentStudentCoop(@PathVariable int id) {
...
+     */
+    @GetMapping("/{id}/upcoming-coops")
+    public List<CoopDto> getUpcomingStudentCoops(@PathVariable int id) {
+        Student s = studentService.getStudentById(id);
+        Set<Coop> coops = s.getCoops();
+        List<CoopDto> result = new ArrayList<>();
+        for (Coop c : coops) {
+            if (c.getStatus() == CoopStatus.FUTURE
+                    || c.getStatus() == CoopStatus.UNDER_REVIEW
+                    || c.getStatus() == CoopStatus.INCOMPLETE)
```
Comment (emmaeagles) [inline Backend/src/main/java/ca/mcgill/cooperator/controller/StudentController.java]: how is incomplete an upcoming coop

Comment (kraglalbert) [reply Backend/src/main/java/ca/mcgill/cooperator/controller/StudentController.java]: I was thinking `INCOMPLETE` meant a student added a new co-op but it got rejected by an Admin; technically it's still upcoming

File: Backend/src/main/java/ca/mcgill/cooperator/dto/CoopDetailsDto.java [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
 
 public class CoopDetailsDto {
     private Integer id;
     private Integer payPerHour;
     private Integer hoursPerWeek;
+
+    @JsonFormat(
+            shape = JsonFormat.Shape.STRING,
+            pattern = "yyyy-MM-dd",
+            timezone = "America/Toronto")
```
Comment (PaulHooley) [inline Backend/src/main/java/ca/mcgill/cooperator/dto/CoopDetailsDto.java]: What purpose do we need timezone for, deadlines?

File: Frontend/src/boot/common.js [hunk 3]
```diff
@@ -39,6 +39,46 @@ const common = {
...
+   */
+  convertSQLDateToReadableString: function(date) {
+    return moment(date)
+      .utc()
+      .format("MMMM Do, YYYY");
+  },
+  /**
+   * Converts a dollar value in cents (e.g. 2000) to a normal dollar
+   * representation (e.g. $20.00)
+   */
```
Comment (PaulHooley) [inline Frontend/src/boot/common.js]: Does this mean everything will be stored in cents. Does it also mean the user will input their cent earnings or will there be a textbox where $20.00 is entered and its converted to cents back and forth between fronend and backend

File: Backend/src/main/java/ca/mcgill/cooperator/controller/StudentController.java (hunk 1 above)
Comment (PaulHooley) [reply Backend/src/main/java/ca/mcgill/cooperator/controller/StudentController.java]: Add Rejected and Failed as enums instead of INCOMPLETE

**Pull request merged**